### ISSUE TITLE
Table doesn't exist

### ISSUE THREAD

**Issue by MightyDrakeC** [stwa/google_addressbook#35]
Brand new Roundcube install. Installed Composer and edited distribution JSON file to add google_addressbook.

When it tried to do its initial "Refresh" it popped up an error about a missing table, that faded quickly.  I pulled this out of the errors file:

> PHP Fatal error:  require_once(): Failed opening required '/home/mightydr/public_html/mytmail/plugins/google_addressbook/../../vendor/autoload.php' (include_path='/home/mightydr/public_html/mytmail/program/lib:.:/usr/lib/php:/usr/local/lib/php:/home/mightydr/pear/php') in /home/mightydr/public_html/mytmail/plugins/google_addressbook/google_func.php on line 10

and

> DB Error: [1146] Table 'mightydr_rcmail1.contacts_google' doesn't exist (SQL Query: DELETE FROM contacts_google WHERE user_id='1') in /home/mightydr/public_html/mytmail/program/lib/Roundcube/rcube_db.php on line 543 (POST /?_task=settings&_action=refresh?_task=&_action=)

Now, when I log in, it says:

> **DATABASE ERROR: CONNECTION FAILED!**
>
>Unable to connect to the database!
>Please contact your server-administrator.

Clicking on Mail takes me to my inbox.

Should I have waited to install this plugin until after I'd run Composer once with the stock JSON file?  Is there some way to initialize it at this point?

Thanks,
Drake

**Comment by stwa** [comment]
Have you created the database tables? In the plugin folder there is a folder named SQL where you can find the schema files.

**Comment by MightyDrakeC** [comment]
I did not see instructions anywhere that spell this out.  Or even
mention it, for that matter.  The impression given from the Plugins page
is that running Composer is sufficient. 

Is there a list of steps spelled out?  I know conceptually what to do,
but I need an example of the syntax. 

Thanks, 

Drake Christensen 

On 2015-12-04 09:31, Stefan Wagner wrote: 

> Have you created the database tables? In the plugin folder there is a folder named SQL where you can find the schema files. 
> 
> --
> Reply to this email directly or view it on GitHub [1].
 

Links:
------
[1]
https://github.com/stwa/google_addressbook/issues/35#issuecomment-161996203

**Comment by stwa** [comment]
It is written in the README.md - "Installation". You're right, there are no explicit steps you can just copy. The steps depend on your database.
Let me know if I can help you any further, and feel free to contribute the necessary steps to the plugins README.

**Comment by MightyDrakeC** [comment]
Ah, thanks.  That's the clue I needed. 

My webhost has phpadmin.  So, I just had to download the SQL file to my
computer in order to put it where the web app could find it, for import.


The Roundcube people should really add a mention of readme files to the
main plugin page.  For someone like me who last installed Roundcube
about five years ago, the whole Composer thing is new, and kind of hides
things like that.  Used to be, to install a plugin, I had to download
the .zip myself, and the readme would jump out at me.  With the new
system, I didn't go looking for it. 

You don't have control of their page, so I would recommend that you add
a line on the your plugin page mentioning the readme file. 

Another alternative would be to try to find a way in the install process
to import the table automagically.  That would be a lot more work than
just adding a line on the website :-) 

Looks like the plugin is up and running.  I appreciate the work you did
on this.  It's extremely useful, and works seamlessly.  Good job. 

Drake 

On 2015-12-07 02:22, Stefan Wagner wrote: 

> It is written in the README.md - "Installation". You're right, there are no explicit steps you can just copy. The steps depend on your database.
> Let me know if I can help you any further, and feel free to contribute the necessary steps to the plugins README. 
> 
> --
> Reply to this email directly or view it on GitHub [1].
 

Links:
------
[1]
https://github.com/stwa/google_addressbook/issues/35#issuecomment-162446366
